tests/unit/lastuser_core/test_model_user_User.py

```diff
@@ -0,0 +1,466 @@
+# -*- coding: utf-8 -*-
+
+from lastuserapp import db
+import lastuser_core.models as models
+from .test_db import TestDatabaseFixture
+from sqlalchemy.orm.collections import InstrumentedList
+from datetime import datetime, timedelta
+import time
+
+
+class TestUser(TestDatabaseFixture):
+
+    def test_User(self):
+        """
+        Test for creation of user object from User model
+        """
+        user = models.User(username=u"lena", fullname=u"Lena Audrey Dachshund")
+        db.session.add_all([user])
+        db.session.commit()
+        lena = models.User.query.filter_by(username=u"lena").first()
+        self.assertIsInstance(lena, models.User)
+        self.assertEqual(user.username, u"lena")
+        self.assertEqual(user.fullname, u"Lena Audrey Dachshund")
+
+    def test_user_is_valid_username(self):
+        """
+        Test to check if given is a valid username associated with the user
+        """
+        crusoe = self.fixtures.crusoe
+        # scenario 1: not a valid username
+        number_one = models.User(username=u'number1', fullname=u'Number One')
+        self.assertFalse(number_one.is_valid_username(u'Number1'))
+        # scenario 2: a valid username but not the username of instance passed
```

Comment:
There are three scenarios for this test

- The username has valid characters (which is technically a proxy for the `valid_username` function in coaster.utils)
- The username has invalid characters (which is technically a proxy for the `valid_username` function in coaster.utils)
- The username is invalid because it has already been used by a user(that isn't the current user)
- The username is invalid because it has already been used by an organization
- The user is valid because it satisfies all the above conditions